### DISCUSSION TITLE
No way to go to the position where last gd (coc-definition) was used?

### DISCUSSION THREAD

**Discussion by vitaly-zdanevich** [neoclide/coc.nvim#3929] [Q&A] [answered]
Hi, in [Tsuquyomi](https://github.com/Quramy/tsuquyomi/blob/8f6e2464171c5367867959093aa4b9c99b636a57/README.md?plain=1#L123) it was so useful feature.

**Comment by NoahTheDuke** [comment, marked as answer]
You can achieve this with `<C-o>`, which uses the jumplist to go backwards. [Stack Overflow answer](https://stackoverflow.com/a/3985410/3023252) about this, or you can search `:help jumplist` to read more about it in vim.

**Reply by vitaly-zdanevich** [reply]
`<C-o>` firstly jump to the first line of buffer. Even with `<C-o>` I can switch to a few buffers, but then would be useful to teleport to the place before coc-definition.

**Reply by NoahTheDuke** [reply]
I'm not sure what mappings you have set up, but this is straight from the [vim 8 jumplist docs](https://vimhelp.org/motion.txt.html#jumplist):

> Jumps are remembered in a jump list.  With the CTRL-O and CTRL-I command you
> can go to cursor positions before older jumps, and back again.  Thus you can
> move up and down the list.  There is a separate jump list for each window.
> The maximum number of entries is fixed at 100.

You press `gd` (or whatever your keymap is) to Go to Definition and then press `<C-o>` (that's the letter `o`, not the number `0`) to jump back to the usage. If this doesn't work, I recommend executing `:verbose map <C-o>` to see what's been mapped in that place.

**Comment by fannheyward** [comment]
You can `set tagfunc=CocTagFunc`, then use `C-t` to jump back. Make sure your vim supports `tagfunc`. Check `:h CocTagFunc`.

**Reply by vitaly-zdanevich** [reply]
`:echo exists('&tagfunc')` return `1`.

Added `set tagfunc=CocTagFunc`
Moved to a function with `gd`
`C-t` return `E555: at bottom of tag stack`.

**Reply by jinying-che** [reply]
@vitaly-zdanevich you may use `Ctrl + ]` to go definition and  `Ctrl + t` go back as mentioned in doc:

```
CocTagFunc({pattern}, {flags}, {info})			CocTagFunc()

	Used for vim's 'tagfunc' option, to make tag search by |CTRL-]| use
	coc.nvim as provider, tag search would be performed when no result
	from coc.nvim.

	Make sure your vim support 'tagfunc' by

	:echo exists('&tagfunc')
```

By the way, make sure the `Ctrl + ]` isn't mapped to `gd` (that's the case for me, I clean the mapping, and everything is working fine then)

**Reply by wurikiji** [reply]
This works perfectly.
